basyx.submodelrepository/basyx.submodelrepository-feature-authorization/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/feature/authorization/rbac/SubmodelTargetPermissionVerifier.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (C) 2023 the Eclipse BaSyx Authors
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * 
+ * SPDX-License-Identifier: MIT
+ ******************************************************************************/
+
+package org.eclipse.digitaltwin.basyx.submodelrepository.feature.authorization.rbac;
+
+import org.eclipse.digitaltwin.basyx.authorization.rbac.RbacRule;
+import org.eclipse.digitaltwin.basyx.authorization.rbac.TargetPermissionVerifier;
+import org.eclipse.digitaltwin.basyx.submodelrepository.feature.authorization.SubmodelTargetInformation;
+
+/**
+ * Verifies the {@link SubmodelTargetInformation} against the {@link RbacRule}
+ * 
+ * @author danish
+ */
+public class SubmodelTargetPermissionVerifier implements TargetPermissionVerifier<SubmodelTargetInformation> {
+
+	private static final String ALL_ALLOWED_WILDCARD = "*";
+
+	@Override
+	public boolean isVerified(RbacRule rbacRule, SubmodelTargetInformation targetInformation) {
+		
+		String submodelId = targetInformation.getSubmodelId();
+		String submodelElementIdShortPath = targetInformation.getSubmodelElementIdShortPath();
+		
+		SubmodelTargetInformation rbacRuleAasTargetInformation = (SubmodelTargetInformation) rbacRule.getTargetInformation();
+
```

Comment:
rbacRuleAasTargetInformation -> rbacRuleSubmodelTargetInformation

Reply:
good point :D i don't remember if i really got what you ment, when we spoke, but i would have wrote this comment right now, so good point

Reply:
I have corrected this.